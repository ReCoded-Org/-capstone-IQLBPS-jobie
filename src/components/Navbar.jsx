import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faMagnifyingGlass, faGlobe, faBars} from '@fortawesome/free-solid-svg-icons'

const Navbar = () => {

    return (
        <nav className="flex justify-center space-x-4 float-center ">

        <div className='flex justify-evenly items-center  md:flex'>
            <div>
            <img src='/logo.svg' alt='logo' className='h-8 md:flex md:h-12'/>
            </div>
            <div  className="hidden md:flex">
      <a href="/dashboard" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">Home</a>
      <a href="/team" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">About</a>
      <a href="/projects" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">Roadmap</a>
      <a href="/reports" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">Jobs</a>
      <a href="/reports" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">Contact</a>
      <a href="/reports" className="font-small px-1 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900">  <FontAwesomeIcon icon= {faGlobe }/>
</a>

      <div className='w-40 flex items-center md: flex'>
      <input
      
                    type="text"
                    className=" block w-full px-4 py-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40 w-full"
                    placeholder="Search..."/>
                    <button type="button" className='-ml-6'>
                     <FontAwesomeIcon icon= {faMagnifyingGlass }/>
                    </button>
                    
                    </div>

                  <button type="button" className="flex bg-red-400 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full">Login</button>


    </div>

    {/* mobile button goes here */}
 <div  className='md:hidden flex items-center'>
        <button type='button' className='mobile-menu-button'>
    <FontAwesomeIcon icon= {faBars}/>
    </button>
    </div>
    
    </div>
    {/* mobile menu goes here */}
    <div className='mobile-menu hidden md:hidden'>
        
mobile menu
{/* <a href='/dashboard' className='block py-2 px-4 text-sm hover:bg-gray-200'>Home</a>
<a href='/team' className='block py-2 px-4 text-sm hover:bg-gray-200'>About</a>
<a href='/projects' className='block py-2 px-4 text-sm hover:bg-gray-200'>Roadmap</a>
<a href='/reports' className='block py-2 px-4 text-sm hover:bg-gray-200'>Jobs</a>
<a href='/reports' className='block py-2 px-4 text-sm hover:bg-gray-200'>Contact</a>
<a href='/reports' className='block py-2 px-4 text-sm hover:bg-gray-200'>Home</a> */}

 </div>

 
    </nav>



    )
}

export default Navbar