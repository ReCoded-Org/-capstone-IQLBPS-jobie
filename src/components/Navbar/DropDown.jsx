import { Fragment } from "react";
import { Menu, Transition } from "@headlessui/react";

function DropDown() {
  return (
    <Menu
      as="div"
      data-testid="filter-button"
      className="relative inline-block text-left"
    >
      <div>
        <Menu.Button
          data-testid="filter-botton-toggle"
          className="bg-red-400 hover:bg-blue-700 text-white font-bold py-2 px-9 mx-4 rounded-full"
        >
          {" "}
          Account
        </Menu.Button>
      </div>

      <Transition
        as={Fragment}
        enter="transition ease-out duration-100"
        enterFrom="transform opacity-0 scale-95"
        enterTo="transform opacity-100 scale-100"
        leave="transition ease-in duration-75"
        leaveFrom="transform opacity-100 scale-100"
        leaveTo="transform opacity-0 scale-95"
      >
        <Menu.Items className="absolute left-0 md:right-0 z-10 mt-2 w-40 origin-center md:origin-top-right rounded-md bg-accent shadow-lg ring-1 ring-white ring-opacity-5 focus:outline-none">
          <div>
            <Menu.Item>
              {({ active }) => (
                <div
                  data-testid="language-button-option"
                  aria-hidden="true"
                  className={`${
                    active
                      ? "bg-white text-secondary rounded"
                      : "text-black-500"
                  }`}
                >
                  <h3>View Profile</h3>
                </div>
              )}
            </Menu.Item>
            <Menu.Item>
              {({ active }) => (
                <div
                  data-testid="language-button-option"
                  aria-hidden="true"
                  className={`${
                    active
                      ? "bg-white text-secondary rounded"
                      : "text-black-500"
                  }`}
                >
                  <h3>Edit Profile</h3>
                </div>
              )}
            </Menu.Item>
            <Menu.Item>
              {({ active }) => (
                <div
                  data-testid="language-button-option"
                  aria-hidden="true"
                  className={`${
                    active
                      ? "bg-white text-secondary rounded"
                      : "text-black-500"
                  }`}
                >
                  <h3>signout</h3>
                </div>
              )}
            </Menu.Item>
          </div>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}

export default DropDown;
