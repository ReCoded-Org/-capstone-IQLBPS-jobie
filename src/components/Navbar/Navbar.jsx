/* eslint-disable */
/* eslint-disable no-console, no-control-regex*/
import React, { useState } from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMagnifyingGlass, faGlobe } from "@fortawesome/free-solid-svg-icons";
import Button from "./Button";
import Dropdown from "./DropDown";
import { useTranslation } from "react-i18next";

const Navbar = () => {
  const Links = [
    { name: "Home", link: "/" },
    { name: "About", link: "/about-us" },
    { name: "Roadmap", link: "/roadmap" },
    { name: "Jobs", link: "/jobs" },
    { name: "Contact", link: "/contact" },
  ];
  const isAuthenticated = true;
  const [open, setOpen] = useState(false);
  const [openDropDown, setOpenDropDown] = useState(false);
  const [isEnglish, setIsEnglish] = useState(true);

  const { i18n } = useTranslation();

  const handleTranslate = () => {
    setIsEnglish(!isEnglish);
    i18n.changeLanguage(isEnglish ? "en" : "ar");
  };
  return (
    <div className="bg-white lg:flex-nowrap md:flex-wrap shadow-md max-w-10 sticky top-0 left-0 z-10 md:flex items-center  justify-between py-4 md:px-10 px-7">
      <img
        src="/logo.svg"
        alt="logo"
        className="h-8 md:flex md:h-12 cursor-pointer"
      />
      <div
        onClick={() => setOpen(!open)} //set open if close.
        className="text-3xl absolute right-8 top-6 cursor-pointer md:hidden object-scale-down"
      >
        <ion-icon name={open ? "close" : "menu"} />
      </div>

      <ul
        className={`md:flex md:items-center md:justify-item-ends md:pb-0 pb-12 absolute md:static bg-white md:z-auto z-[-1] md:pl-0 pl-9 transition-all duration-500 ease-in ${
          open ? "top-20" : "top-[-490px]"
        }`}
      >
        {Links.map((link) => (
          <li
            key={link.name}
            className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
          >
            <Link
              to={link.link}
              className="text-gray-800 hover:text-gray-400 duration-500"
            >
              {link.name}
            </Link>
          </li>
        ))}

        <span onClick={handleTranslate} className="mx-3">
          <FontAwesomeIcon icon={faGlobe} className="mx-5" />
          {isEnglish ? "En" : "Ku"}
        </span>
        <div className="lg:w-50 md:w-45 flex items-center">
          <input
            type="text"
            className="block px-4 md:my-0 my-5 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40 w-full"
            placeholder="Search..."
          />
          <button type="button" className="-ml-6">
            <FontAwesomeIcon icon={faMagnifyingGlass} />
          </button>
        </div>
        {isAuthenticated ? <Dropdown /> : <Button>Login</Button>}
      </ul>
    </div>
  );
};

export default Navbar;
