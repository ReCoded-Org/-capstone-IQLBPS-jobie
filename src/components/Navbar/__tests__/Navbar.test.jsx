import renderer from "react-test-renderer";
import Navbar from "../Navbar.jsx";
import React from "react";

it("renders Meet Our Team component correctly", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
