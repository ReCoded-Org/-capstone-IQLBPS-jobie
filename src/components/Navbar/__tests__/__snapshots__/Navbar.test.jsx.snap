// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Meet Our Team component correctly 1`] = `
<div
  className="bg-white lg:flex-nowrap md:flex-wrap shadow-md max-w-10 sticky top-0 left-0 z-10 md:flex items-center  justify-between py-4 md:px-10 px-7"
>
  <img
    alt="logo"
    className="h-8 md:flex md:h-12 cursor-pointer"
    src="/logo.svg"
  />
  <div
    className="text-3xl absolute right-8 top-6 cursor-pointer md:hidden object-scale-down"
    onClick={[Function]}
  >
    <ion-icon
      name="menu"
    />
  </div>
  <ul
    className="md:flex md:items-center md:justify-item-ends md:pb-0 pb-12 absolute md:static bg-white md:z-auto z-[-1] md:pl-0 pl-9 transition-all duration-500 ease-in top-[-490px]"
  >
    <li
      className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
    >
      <a
        className="text-gray-800 hover:text-gray-400 duration-500"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
    >
      <a
        className="text-gray-800 hover:text-gray-400 duration-500"
        href="/about-us"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li
      className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
    >
      <a
        className="text-gray-800 hover:text-gray-400 duration-500"
        href="/roadmap"
        onClick={[Function]}
      >
        Roadmap
      </a>
    </li>
    <li
      className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
    >
      <a
        className="text-gray-800 hover:text-gray-400 duration-500"
        href="/jobs"
        onClick={[Function]}
      >
        Jobs
      </a>
    </li>
    <li
      className="md:ml-8 lg:text-xl md:text-sm lg:my-0 my-7"
    >
      <a
        className="text-gray-800 hover:text-gray-400 duration-500"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </li>
    <span
      className="mx-3"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-globe mx-5"
        data-icon="globe"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 256c0 22.2-1.2 43.6-3.3 64H163.3c-2.2-20.4-3.3-41.8-3.3-64s1.2-43.6 3.3-64H348.7c2.2 20.4 3.3 41.8 3.3 64zm28.8-64H503.9c5.3 20.5 8.1 41.9 8.1 64s-2.8 43.5-8.1 64H380.8c2.1-20.6 3.2-42 3.2-64s-1.1-43.4-3.2-64zm112.6-32H376.7c-10-63.9-29.8-117.4-55.3-151.6c78.3 20.7 142 77.5 171.9 151.6zm-149.1 0H167.7c6.1-36.4 15.5-68.6 27-94.7c10.5-23.6 22.2-40.7 33.5-51.5C239.4 3.2 248.7 0 256 0s16.6 3.2 27.8 13.8c11.3 10.8 23 27.9 33.5 51.5c11.6 26 21 58.2 27 94.7zm-209 0H18.6C48.6 85.9 112.2 29.1 190.6 8.4C165.1 42.6 145.3 96.1 135.3 160zM8.1 192H131.2c-2.1 20.6-3.2 42-3.2 64s1.1 43.4 3.2 64H8.1C2.8 299.5 0 278.1 0 256s2.8-43.5 8.1-64zM194.7 446.6c-11.6-26-20.9-58.2-27-94.6H344.3c-6.1 36.4-15.5 68.6-27 94.6c-10.5 23.6-22.2 40.7-33.5 51.5C272.6 508.8 263.3 512 256 512s-16.6-3.2-27.8-13.8c-11.3-10.8-23-27.9-33.5-51.5zM135.3 352c10 63.9 29.8 117.4 55.3 151.6C112.2 482.9 48.6 426.1 18.6 352H135.3zm358.1 0c-30 74.1-93.6 130.9-171.9 151.6c25.5-34.2 45.2-87.7 55.3-151.6H493.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      En
    </span>
    <div
      className="lg:w-60 md:w-45 flex items-center"
    >
      <input
        className="block px-4 md:my-0 my-5 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40 w-full"
        placeholder="Search..."
        type="text"
      />
      <button
        className="-ml-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
    <div
      className="relative inline-block text-left"
      data-headlessui-state=""
      data-testid="filter-button"
    >
      <div>
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="bg-red-400 hover:bg-blue-700 text-white font-bold py-2 px-5 mx-3 rounded-full"
          data-headlessui-state=""
          data-testid="filter-botton-toggle"
          id="headlessui-menu-button-:r0:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
           
          Account
        </button>
      </div>
    </div>
  </ul>
</div>
`;
